qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -76,19 +80,58 @@ def run(self, qobj: Qobj, job_name: Optional[str] = None) -> IBMQJob:
 
         Returns:
             IBMQJob: an instance derived from BaseJob
+
+        Raises:
+            SchemaValidationError: If the job validation fails.
+            IBMQBackendError: If an unexpected error occurred while submitting
+                the job.
         """
         # pylint: disable=arguments-differ
-        use_websockets = True
-        if self._credentials.proxies:
-            # Disable using websockets through proxies.
-            use_websockets = False
-        job = IBMQJob(
-            self, None, self._api,
-            qobj=qobj,
-            use_object_storage=getattr(self.configuration(), 'allow_object_storage', False),
-            use_websockets=use_websockets)
-        job.submit(job_name=job_name)
 
+        validate_qobj_against_schema(qobj)
+        return self._submit_job(qobj, job_name)
+
+    def _submit_job(self, qobj: Qobj, job_name: Optional[str] = None) -> IBMQJob:
+        """Submit qobj job to IBM-Q.
+        Args:
+            qobj (Qobj): description of job.
+            job_name (str): custom name to be assigned to the job. This job
+                name can subsequently be used as a filter in the
+                ``jobs()`` function call. Job names do not need to be unique.
+
+        Returns:
+            IBMQJob: an instance derived from BaseJob
+
+        Raises:
+            IBMQBackendError: If an unexpected error occurred while submitting
+                the job.
+        """
+        try:
+            submit_info = self._api.job_submit(
+                backend_name=self.name(),
+                qobj_dict=qobj.to_dict(),
+                use_object_storage=getattr(self.configuration(), 'allow_object_storage', False),
+                job_name=job_name)
+        except ApiError as ex:
+            raise IBMQBackendError('Error submitting job: {}'.format(str(ex)))
+
+        # Error in the job after submission:
+        # Transition to the `ERROR` final state.
+        if 'error' in submit_info:
+            raise IBMQBackendError('Error submitting job: {}'.format(str(submit_info['error'])))
+
+        # Submission success.
+        submit_info.update({
+            'backend_obj': self,
+            'api': self._api,
+            'qobj': qobj
+        })
+        try:
+            job = IBMQJob.from_dict(submit_info)
+        except ModelValidationError as err:
+            raise IBMQBackendError('Unexpected return value from the server when '
+                                   'submitting job: {}'.format(str(err)))
+        Publisher().publish("ibmq.job.start", job)
```

Comment:
Can you add a note to the docstring about the event, moving the previous one at https://github.com/Qiskit/qiskit-ibmq-provider/blob/master/qiskit/providers/ibmq/job/ibmqjob.py#L427?